model/DeliveryExecutionGetter.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA;
+ */
+
+namespace oat\taoLtiConsumer\model;
+
+use common_exception_NotFound;
+use core_kernel_classes_Resource;
+use oat\generis\model\OntologyAwareTrait;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoDelivery\model\execution\DeliveryExecutionInterface;
+use oat\taoDelivery\model\execution\ServiceProxy;
+
+class DeliveryExecutionGetter extends ConfigurableService
+{
+    use OntologyAwareTrait;
+
+    /**
+     * Duplicated to not to create direct dependency between extensions
+     * @see \oat\taoDeliverConnect\model\delivery\factory\RemoteDeliveryFactory::PROPERTY_TENANT_ID
+     */
+    public const PROPERTY_TENANT_ID = 'http://www.tao.lu/Ontologies/taoDeliverConnect.rdf#TenantId';
+
+    /**
+     * Due to multiple implementation of DE storages it's difficult to check if DE exists
```

Comment:
Why can't this https://github.com/oat-sa/extension-tao-delivery/blob/develop/model/execution/KVDeliveryExecution.php#L191 be used?

Reply:
Fixed